lightning/src/ln/channel.rs

```diff
@@ -1160,6 +1160,45 @@ impl<'a, SP: Deref> ChannelPhase<SP> where
 	}
 }
 
+/// A top-level channel struct, containing a channel phase
+pub(super) struct ChannelWrapper<SP: Deref> where SP::Target: SignerProvider {
+	phase: ChannelPhase<SP>,
+}
+
+impl<'a, SP: Deref> ChannelWrapper<SP> where SP::Target: SignerProvider {
+	pub fn new(phase: ChannelPhase<SP>) -> Self {
+		Self { phase }
+	}
+
+	pub fn phase(&self) -> &ChannelPhase<SP> {
+		&self.phase
+	}
+
+	pub fn phase_mut(&mut self) -> &mut ChannelPhase<SP> {
+		&mut self.phase
+	}
+
+	pub fn phase_take(self) -> ChannelPhase<SP> {
+		self.phase
+	}
+
+	pub fn get_funded_channel(&self) -> Option<&Channel<SP>> {
```

Comment:
We aren't consistent about this, but no need to use `get_` prefix for accessor methods.